bug in neg_binomial_2_log_glm_lpmf.hpp is breaking stan/math

#### Summary:
Trying to run a simple hello_world example from the Readme on a fresh clone of the `develop` branch leads to errors and failure of the program. The errors originate from the file `neg_binomila_2_log_glm_lpmf.hpp`.
By commenting out line 292 in `~/stan-dev/math/stan/math/prim/mat.hpp` like so
`//#include <stan/math/prim/mat/prob/neg_binomial_2_log_glm_lpmf.hpp>`
, the program is able to run as expected.


#### Description:
When trying to run the hello world example from the Readme of `stan-dev/math` (printing the output of `normal_log(1,2,3)`, erros are raised by `neg_binomila_2_log_glm_lpmf.hpp` and execution fails. Here is a print out of the error logs:
```
| ~/github_projects/stan-dev/math/hello_world @ (develop)
| => clang++ -I ~/github_projects/stan-dev/math -I ~/github_projects/stan-dev/math/lib/eigen_3.3.3/ -I ~/github_projects/stan-dev/math/lib/boost_1.64.0/ -I ~/github_projects/stan-dev/math/lib/cvodes_2.9.0/include hello_world.cpp
In file included from hello_world.cpp:1:
In file included from /Users/youcefmsaid/github_projects/stan-dev/math/stan/math.hpp:4:
In file included from /Users/youcefmsaid/github_projects/stan-dev/math/stan/math/rev/mat.hpp:12:
In file included from /Users/youcefmsaid/github_projects/stan-dev/math/stan/math/prim/mat.hpp:249:
In file included from /Users/youcefmsaid/github_projects/stan-dev/math/stan/math/prim/mat/prob/bernoulli_logit_glm_lpmf.hpp:15:
/Users/youcefmsaid/github_projects/stan-dev/math/stan/math/prim/scal/fun/size_zero.hpp:30:31: warning: variadic templates are
      a C++11 extension [-Wc++11-extensions]
template <typename T, typename... Ts>
                              ^
/Users/youcefmsaid/github_projects/stan-dev/math/stan/math/prim/scal/fun/size_zero.hpp:31:31: warning: rvalue references are a
      C++11 extension [-Wc++11-extensions]
inline bool size_zero(T& x, Ts&&... xs) {
                              ^
In file included from hello_world.cpp:1:
In file included from /Users/youcefmsaid/github_projects/stan-dev/math/stan/math.hpp:4:
In file included from /Users/youcefmsaid/github_projects/stan-dev/math/stan/math/rev/mat.hpp:12:
In file included from /Users/youcefmsaid/github_projects/stan-dev/math/stan/math/prim/mat.hpp:292:
/Users/youcefmsaid/github_projects/stan-dev/math/stan/math/prim/mat/prob/neg_binomial_2_log_glm_lpmf.hpp:116:39: error:
      expected expression
      = theta_dbl.binaryExpr(log_phi, [](const T_partials_return& xx,
                                      ^
/Users/youcefmsaid/github_projects/stan-dev/math/stan/math/prim/mat/prob/neg_binomial_2_log_glm_lpmf.hpp:126:21: error:
      expected expression
                    [](const T_partials_return& xx) { return lgamma(xx); })
                    ^
/Users/youcefmsaid/github_projects/stan-dev/math/stan/math/prim/mat/prob/neg_binomial_2_log_glm_lpmf.hpp:132:18: error:
      expected expression
                 [](const T_partials_return& xx, const T_partials_return& yy) {
                 ^
/Users/youcefmsaid/github_projects/stan-dev/math/stan/math/prim/mat/prob/neg_binomial_2_log_glm_lpmf.hpp:136:20: error:
      expected expression
                   [](const T_partials_return& xx) { return lgamma(xx); }))
                   ^
/Users/youcefmsaid/github_projects/stan-dev/math/stan/math/prim/mat/prob/neg_binomial_2_log_glm_lpmf.hpp:146:21: error:
      expected expression
                    [](const T_partials_return& xx) { return lgamma(xx); })
                    ^
/Users/youcefmsaid/github_projects/stan-dev/math/stan/math/prim/mat/prob/neg_binomial_2_log_glm_lpmf.hpp:178:18: error:
      expected expression
                 [](const T_partials_return& xx) { return digamma(xx); })
                 ^
/Users/youcefmsaid/github_projects/stan-dev/math/stan/math/prim/mat/prob/neg_binomial_2_log_glm_lpmf.hpp:180:18: error:
      expected expression
                 [](const T_partials_return& xx) { return digamma(xx); }))
                 ^
2 warnings and 7 errors generated.
```


#### Reproducible Steps:

1. Clone `develop` branch
2. create hello_world folder and hello_world.cpp file with the content
```
#include <stan/math.hpp>
#include <iostream>

int main() {
  std::cout << "log normal(1 | 2, 3)=" 
            << stan::math::normal_log(1, 2, 3) 
            << std::endl;
}
```

3.  in the shell run the clang++ command with the arguments as decribed in the Readme. You should get the errors printed above.
4. Comment out line 292 in `~/stan-dev/math/stan/math/prim/mat.hpp` and run step 3 again. The program should succeed.


#### Current Version:
develop

--- a/README.md
+++ b/README.md
@@ -37,7 +37,7 @@ If this is in the file `/path/to/foo/foo.cpp`, then you can compile and run this
 
 ```
 > cd /path/to/foo
-> clang++ -I /path/to/stan-math -I /path/to/Eigen -I /path/to/boost foo.cpp  -I /path/to/cvodes
+> clang++ -I /path/to/stan-math -I /path/to/Eigen -I /path/to/boost -I /path/to/cvodes -std=c++11 foo.cpp
 > ./a.out
 log normal(1 | 2, 3)=-2.07311
 ```
@@ -52,7 +52,7 @@ The `-I` includes provide paths pointing to the four necessary includes:
 Note that the paths should *not* include the final directories `stan`, `Eigen`, or `boost` on the paths.  An example of a real instantiation:
 
 ```
-clang++ -I ~/stan-dev/math -I ~/stan-dev/math/lib/eigen_3.3.3/ -I ~/stan-dev/math/lib/boost_1.64.0/ -I ~/stan-dev/math/lib/cvodes_2.9.0/include foo.cpp
+clang++ -I ~/stan-dev/math -I ~/stan-dev/math/lib/eigen_3.3.3/ -I ~/stan-dev/math/lib/boost_1.64.0/ -I ~/stan-dev/math/lib/cvodes_2.9.0/include -std=c++11 foo.cpp
 ```
 
 The following directories all exist below the links given to `-I`: `~/stan-dev/math/stan` and `~/stan-dev/math/lib/eigen_3.3.3/Eigen` and `~stan-dev/math/lib/boost_1.64.0/boost` and `~stan-dev/math/lib/cvodes_2.9.0/include`.

--- a/README.md
+++ b/README.md
@@ -37,7 +37,7 @@ If this is in the file `/path/to/foo/foo.cpp`, then you can compile and run this
 
 ```
 > cd /path/to/foo
-> clang++ -I /path/to/stan-math -I /path/to/Eigen -I /path/to/boost -I /path/to/cvodes -std=c++11 foo.cpp
+> clang++ -std=c++11 -I /path/to/stan-math -I /path/to/Eigen -I /path/to/boost -I /path/to/cvodes foo.cpp
 > ./a.out
 log normal(1 | 2, 3)=-2.07311
 ```
@@ -52,7 +52,7 @@ The `-I` includes provide paths pointing to the four necessary includes:
 Note that the paths should *not* include the final directories `stan`, `Eigen`, or `boost` on the paths.  An example of a real instantiation:
 
 ```
-clang++ -I ~/stan-dev/math -I ~/stan-dev/math/lib/eigen_3.3.3/ -I ~/stan-dev/math/lib/boost_1.64.0/ -I ~/stan-dev/math/lib/cvodes_2.9.0/include -std=c++11 foo.cpp
+clang++ -std=c++11 -I ~/stan-dev/math -I ~/stan-dev/math/lib/eigen_3.3.3/ -I ~/stan-dev/math/lib/boost_1.64.0/ -I ~/stan-dev/math/lib/cvodes_2.9.0/include foo.cpp
 ```
 
 The following directories all exist below the links given to `-I`: `~/stan-dev/math/stan` and `~/stan-dev/math/lib/eigen_3.3.3/Eigen` and `~stan-dev/math/lib/boost_1.64.0/boost` and `~stan-dev/math/lib/cvodes_2.9.0/include`.

